rbd.c

```diff
@@ -260,22 +260,31 @@ static void tcmu_rbd_detect_device_class(struct tcmu_device *dev)
 	}
 
 	ret = rados_mon_command(state->cluster, (const char **)mon_cmd_bufs, 1,
-				"", 0, &mon_buf, &mon_buf_len,
+				"", 0, &temp_mon_buf, &temp_mon_buf_len,
 				&mon_status_buf, &mon_status_buf_len);
 	free(mon_cmd_bufs[0]);
-	if (ret < 0 || !mon_buf) {
+	if (ret < 0 || !temp_mon_buf) {
 		tcmu_dev_warn(dev, "Could not retrieve pool crush rule (Err %d)\n",
 			      ret);
 		return;
 	}
 	rados_buffer_free(mon_status_buf);
 
 	/* expected JSON output: "{..."crush_rule":"<rule name>"...}" */
-	mon_buf[mon_buf_len] = '\0';
```

Comment:
Nit: why not just set ```mon_buf[mon_buf_len - 1] = '\0';``` and test for ```mon_buf_len == 0``` on line 266 instead of adding all this extra code? It's in JSON format so there is plenty of extra characters that won't be needed.

Reply:
think you for your nit.yes, you are right,and I don't think it will affect the performance. I don't know if it fits our coding habits to overlay a character. If it fits, your proposal is very good, and I'll update the code according to your advice.

Reply:
I'll leave it for Mike to decide, but these buffers are now owned by the caller and are already sliced up during processing. Thanks for the catch.

Reply:
@letterwuyu, do what dillaman suggested.

When I made my review comment I had thought there were going to be valid non null characters up to the end of the buffer and we needed to add a null after what was returned.

Reply:
ok, I have update this pr.